### PR TITLE
cleanup: grant the correct permissions when creating a file

### PR DESCRIPTION
**Pull request by chaosi-zju**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

cleanup: grant the correct permissions when creating a file

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: hack/tools/lifted-gen/lifted-gen.go [hunk 1]
```diff
@@ -205,7 +207,7 @@ func (a *analyzer) parseComments(comments []*ast.CommentGroup) (item item, ok bo
 }
 
 func (a *analyzer) dump(path string) {
-	file, err := os.Create(path)
+	file, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR, util.DefaultFilePerm)
```
Comment (zhzhuang-zju) [inline hack/tools/lifted-gen/lifted-gen.go:210]: The `path` will contain the directory of the file, please help to confirm this:
- If the directory does not exist, will `os.OpenFile` also create the directory?
- How to control the permissions of the directory

Comment (chaosi-zju) [reply hack/tools/lifted-gen/lifted-gen.go:210]: The logic here is consistent with the previous, confirmed no problem.

Comment (zhzhuang-zju) [reply hack/tools/lifted-gen/lifted-gen.go:210]: thanks
/lgtm

**Comment by zhzhuang-zju** [comment]
cc @RainbowMango

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**